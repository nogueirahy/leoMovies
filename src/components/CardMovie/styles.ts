import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    marginHorizontal: 12,
    marginBottom: 8,
  },
  chip: {
    position: "absolute",
    backgroundColor: "rgba(0, 0, 0, 0.4)",
    right: "0%",
  },

  title: {
    fontWeight: "bold",
    textAlign: "center",
    letterSpacing: 0.3,
  },
  subtitle: {
    textAlign: "center",
  },
});

export const themeChip = {
  roundness: 4,
  colors: {
    disabled: "yellow",
  },
};
