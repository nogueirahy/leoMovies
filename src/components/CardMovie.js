import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import { Card, Chip, Text } from 'react-native-paper';
import { CardMovieStyle } from './styles';
import { themeChip } from './styles/CardMovieStyle';

const propTypes = {
  onPress: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  voteAverage: PropTypes.number.isRequired,
  releaseDate: PropTypes.string.isRequired,
};

function CardMovie({
  title, imageUrl, voteAverage, releaseDate, onPress,
}) {
  return (
    <Card elevation={4} style={CardMovieStyle.container} onPress={onPress}>
      <Card.Cover source={{ uri: imageUrl }} />
      <Chip style={CardMovieStyle.chip} theme={themeChip} icon="star" disabled>
        {voteAverage}
      </Chip>
      <Card.Content>
        <View>
          <Text style={CardMovieStyle.title} numberOfLines={1} ellipsizeMode="tail">
            {title}
          </Text>
          <Text style={CardMovieStyle.subtitle} numberOfLines={1} ellipsizeMode="tail">
            {releaseDate}
          </Text>
        </View>
      </Card.Content>
    </Card>
  );
}

CardMovie.propTypes = propTypes;
export default CardMovie;
