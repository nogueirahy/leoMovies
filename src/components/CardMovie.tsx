import React from "react";
import { View } from "react-native";
import { Card, Chip, Text } from "react-native-paper";

import { CardMovieStyle } from "./styles";
import { themeChip } from "./styles/CardMovieStyle";

interface IProps {
  title: string;
  imageUrl: string;
  voteAverage: string;
  releaseDate: string;
  onPress: () => void;
}

const CardMovie: React.FC<IProps> = ({
  title,
  imageUrl,
  voteAverage,
  releaseDate,
  onPress,
}) => (
  <Card elevation={4} style={CardMovieStyle.container} onPress={onPress}>
    <Card.Cover source={{ uri: imageUrl }} />
    <Chip style={CardMovieStyle.chip} theme={themeChip} icon="star" disabled>
      {voteAverage}
    </Chip>
    <Card.Content>
      <View>
        <Text
          style={CardMovieStyle.title}
          numberOfLines={1}
          ellipsizeMode="tail"
        >
          {title}
        </Text>
        <Text
          style={CardMovieStyle.subtitle}
          numberOfLines={1}
          ellipsizeMode="tail"
        >
          {releaseDate}
        </Text>
      </View>
    </Card.Content>
  </Card>
);

export default CardMovie;
