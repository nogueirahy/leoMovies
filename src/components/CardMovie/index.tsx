import React from "react";
import { Card, Chip } from "react-native-paper";
import styles, { themeChip } from "./styles";

interface IProps {
  imageUrl: string;
  voteAverage: string;
  onPress: () => void;
}

const CardMovie: React.FC<IProps> = ({ imageUrl, voteAverage, onPress }) => (
  <Card elevation={4} style={styles.container} onPress={onPress}>
    <Card.Cover source={{ uri: imageUrl }} style={{ height: 190 }} />
    <Chip style={styles.chip} theme={themeChip} icon="star" disabled>
      {voteAverage}
    </Chip>
  </Card>
);

export default CardMovie;
