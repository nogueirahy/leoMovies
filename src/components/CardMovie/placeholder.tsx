import React from "react";
import { View } from "react-native";
import { Card } from "react-native-paper";
import { Placeholder, PlaceholderMedia, Progressive } from "rn-placeholder";
import styles from "./styles";

export default () => (
  <View style={{ flexDirection: "row" }}>
    <Card elevation={4} style={[styles.container, { height: 190 }]}>
      <Placeholder Animation={Progressive}>
        <PlaceholderMedia style={{ width: 125, height: 190 }} />
      </Placeholder>
    </Card>
  </View>
);
