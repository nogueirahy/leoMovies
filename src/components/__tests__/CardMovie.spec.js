import React from "react";
import { render } from "react-native-testing-library";

import CardMovie from "../CardMovie";

describe("CardMovie", () => {
  it("deve renderizar corretamente", () => {
    const tree = render(<CardMovie />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
