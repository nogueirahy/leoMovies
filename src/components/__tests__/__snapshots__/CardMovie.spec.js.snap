// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardMovie deve renderizar corretamente 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 4,
      "elevation": 4,
      "marginBottom": 18,
      "marginHorizontal": 10,
      "maxWidth": 135,
      "minWidth": 135,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 4,
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#eeeeee",
            "height": 195,
            "overflow": "hidden",
          },
          Object {
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          undefined,
        ]
      }
    >
      <Image
        siblings={
          Array [
            "withTheme(Card.Cover)",
            "withTheme(Chip)",
            "Card.Content",
          ]
        }
        source={
          Object {
            "uri": "https://image.tmdb.org/t/p/w500/dzBtMocZuJbjLOXvrl4zGYigDzh.jpg\\"",
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
              "height": null,
              "justifyContent": "flex-end",
              "padding": 16,
              "resizeMode": "cover",
              "width": null,
            },
            Object {
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
          ]
        }
      />
    </View>
    <View
      index={1}
      siblings={
        Array [
          "withTheme(Card.Cover)",
          "withTheme(Chip)",
          "Card.Content",
        ]
      }
      style={
        Object {
          "backgroundColor": "#212C31",
          "borderColor": "#212C31",
          "borderRadius": 16,
          "borderStyle": "solid",
          "borderWidth": 0.5,
          "bottom": "21%",
          "elevation": 0,
          "position": "absolute",
        }
      }
      total={3}
    >
      <View
        accessibilityRole="button"
        accessibilityStates={
          Array [
            "disabled",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 16,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingHorizontal": 4,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "padding": 4,
                },
                null,
              ]
            }
          >
            <Text />
          </View>
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "Helvetica Neue",
                  "textAlign": "left",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "lineHeight": 24,
                    "marginVertical": 4,
                    "minHeight": 24,
                    "textAlignVertical": "center",
                  },
                  Object {
                    "color": "yellow",
                    "marginLeft": 4,
                    "marginRight": 8,
                  },
                  undefined,
                ],
              ]
            }
          >
            7.7
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 16,
          },
          Object {
            "paddingBottom": 16,
          },
          undefined,
        ]
      }
    >
      <View>
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Helvetica Neue",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Object {
                "fontWeight": "bold",
                "marginTop": 14,
                "textAlign": "center",
              },
            ]
          }
        >
          The Lion King
        </Text>
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Helvetica Neue",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Object {
                "textAlign": "center",
              },
            ]
          }
        >
          Jul 20th 19
        </Text>
      </View>
    </View>
  </View>
</View>
`;
